### PR TITLE
Lab 10

### PR DESCRIPTION
**Pull request by YaroslavKSE**

### CODE REVIEW

File: lecture_notes/10 refactoring/1-not-optimal.js [hunk 1]
```diff
@@ -1,27 +1,38 @@
-'use strict';
+"use strict";
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/1-not-optimal.js:1]: We use double quote

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
- Improve code style in js
- Try to optimize using map instead of for loop
- Don not modify incoming structures like list and point, create new one instead

File: lecture_notes/10 refactoring/1-not-optimal.js (hunk 1 above)
Comment (YaroslavKSE) [reply lecture_notes/10 refactoring/1-not-optimal.js:1]: I use double quote too)

File: lecture_notes/10 refactoring/1-not-optimal.js [hunk 2]
```diff
@@ -1,27 +1,38 @@
...
-      points[i].x += offset.x;
-      points[i].y += offset.y;
-    }
-  });
-  return points;
-};
+rl.question("Enter coordinate X: ", function (X) {
+  rl.question("Enter coordinate Y: ", function (Y) {
+    const x = Number(X);
+    const y = Number(Y);
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/1-not-optimal.js:11]: use parseInt Luke

File: lecture_notes/10 refactoring/1-not-optimal.js [hunk 3]
```diff
@@ -1,27 +1,38 @@
...
+
+    const move = (coords) => (point) => {
+      point.x += coords.x;
+      point.y += coords.y;
+      return point;
+    };
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/1-not-optimal.js:17]: Better to move to file-global scope

File: lecture_notes/10 refactoring/1-not-optimal.js [hunk 4]
```diff
@@ -1,27 +1,38 @@
...
-      let i = points.indexOf(point);
-      points[i] = JSON.parse(point);
-      points[i].x += offset.x;
-      points[i].y += offset.y;
-    }
-  });
-  return points;
-};
+rl.question("Enter coordinate X: ", function (X) {
+  rl.question("Enter coordinate Y: ", function (Y) {
```
Comment (tshemsedinov) [inline lecture_notes/10 refactoring/1-not-optimal.js:9]: Why `function`? use lambdas, X and Y are not camelCase

**Pull request closed without merging**